### PR TITLE
perf(groups): makes group invitations scalable

### PR DESCRIPTION
**Pull request by hypeJunction**
Adds pagination to the groups invitation list
Uses a more efficient query to render the menu item

### CODE REVIEW

File: mod/groups/views/default/groups/invitationrequests.php [hunk 1]
```diff
@@ -46,6 +63,14 @@
 		}
 	}
 	echo '</ul>';
+
+	if (!empty($count)) {
+		echo elgg_view('navigation/pagination', array(
```
Comment (mrclay) [inline mod/groups/views/default/groups/invitationrequests.php]: So icky to have to manage this ourselves. This makes me like the idea of using elgg_list_entities, bat passing in a closure to render items...

Comment (mrclay) [inline mod/groups/views/default/groups/invitationrequests.php]: If this issue isn't pressing, I'd suggest waiting until that's an option.

Comment (hypeJunction) [inline mod/groups/views/default/groups/invitationrequests.php]: I know. I was working on the entity views and came across this overblown limit=>0

Comment (hypeJunction) [inline mod/groups/views/default/groups/invitationrequests.php]: I mean it's not pressing, but I wonder how this never came up as an issue on large sites.

Comment (jdalsem) [inline mod/groups/views/default/groups/invitationrequests.php]: maybe they are using the group_tools plugin :)

**Comment by hypeJunction** [comment]
@mrclay, let's get this in once we get the LGTM's, and I will rebase #7913 if we get the entity view PR in

File: mod/groups/views/default/groups/invitationrequests.php (hunk 1 above)
Comment (jdalsem) [inline mod/groups/views/default/groups/invitationrequests.php]: why not switch to `elgg_view('page/components/list')` here? Then there is no need to draw pagination yourself

Comment (jdalsem) [inline mod/groups/views/default/groups/invitationrequests.php]: my bad... that is currently not possible

**Pull request merged**